src/analysis.zig

```diff
@@ -1446,17 +1446,56 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
             if (std.mem.eql(u8, call_name, "@field")) {
                 if (params.len < 2) return null;
                 const field_param = params[1];
-                if (node_tags[field_param] != .string_literal) return null;
+                const field_name = fld_name_blk: {
+                    switch (node_tags[field_param]) {
+                        .string_literal => {
+                            const field_name = handle.tree.tokenSlice(main_tokens[field_param]);
+                            break :fld_name_blk field_name;
+                        },
+                        .identifier,
+                        .field_access,
+                        => {
+                            var resolved_type = (try analyser.resolveTypeOfNodeInternal(.{
+                                .node = field_param,
+                                .handle = handle,
+                            })) orelse return null;
+                            switch (resolved_type.type.data) {
+                                // const name = "name"
+                                .other => |n| {
+                                    if (resolved_type.handle.tree.nodes.items(.tag)[n] != .string_literal) return null;
+                                    break :fld_name_blk offsets.nodeToSlice(resolved_type.handle.tree, n);
+                                },
+                                // const name: []const u8 = "name";
+                                .pointer => |pointer| {
+                                    if (pointer.elem_ty.type.data != .ip_index) return null;
+                                    var str_node = pointer.elem_ty.type.data.ip_index.node orelse return null;
+                                    str_node += 3; // XXX str_node points to the "u8", + 3 gets "name"
+                                    const elem_ty_tree = pointer.elem_ty.handle.tree;
+                                    if ((str_node > elem_ty_tree.nodes.len - 1) or (elem_ty_tree.nodes.items(.tag)[str_node] != .string_literal)) return null;
+                                    break :fld_name_blk offsets.nodeToSlice(elem_ty_tree, str_node);
+                                },
+                                else => return null,
+                            }
+                        },
+                        else => return null,
+                    }
+                };
```

Comment:
I would suggest to use `resolveVarDeclAlias` instead of  `resolveTypeOfNodeInternal` because you don't search for a type but rather the initialization expression hidden behind some indirection/aliasing that you can undo with `resolveVarDeclAlias`. This also makes in unnecessary to set the node when resolving primitve types.

```suggestion
                var field_name_node: NodeWithHandle = .{ .node = params[1], .handle = handle };

                if (try analyser.resolveVarDeclAlias(field_name_node)) |decl_with_handle| {
                    if (decl_with_handle.decl == .ast_node) {
                        field_name_node = .{
                            .node = decl_with_handle.decl.ast_node,
                            .handle = decl_with_handle.handle,
                        };
                    }
                }
                const string_literal_node = switch (field_name_node.handle.tree.nodes.items(.tag)[field_name_node.node]) {
                    .string_literal => field_name_node.node,
                    .global_var_decl,
                    .local_var_decl,
                    .aligned_var_decl,
                    .simple_var_decl,
                    => blk: {
                        const init_node = field_name_node.handle.tree.fullVarDecl(field_name_node.node).?.ast.init_node;
                        if (field_name_node.handle.tree.nodes.items(.tag)[init_node] != .string_literal) return null;
                        break :blk init_node;
                    },
                    else => return null,
                };

                const field_name_token = field_name_node.handle.tree.nodes.items(.main_token)[string_literal_node];
                const field_name = offsets.tokenToSlice(field_name_node.handle.tree, field_name_token);

```

Reply:
Done